File: src/Feedback.Analyzer.Backend/Feedback.Analyzer.Domain/Entities/PromptExecutionHistory.cs
using System.ComponentModel.DataAnnotations.Schema;
using Feedback.Analyzer.Domain.Common.Entities;

namespace Feedback.Analyzer.Domain.Entities;

/// <summary>
/// Represents the execution history of a prompt.
/// </summary>
public class PromptExecutionHistory : Entity
{ 
    /// <summary>
    /// Gets or sets the unique identifier of the prompt.
    /// </summary>
    public Guid PromptId { get; set; }

    /// <summary>
    /// Gets or sets the result of the prompt execution.
    /// </summary>
    public string? Result { get; set; }

    /// <summary>
    /// Gets or sets the exception encountered during prompt execution, if any.
    /// </summary>
    public string? Exception { get; set; } 
    
    /// <summary>
    /// Gets or sets the time when the prompt was executed.
    /// </summary>
    public DateTimeOffset ExecutionTime { get; set; } 
    
    /// <summary>
    /// Gets or sets the duration of the prompt execution.
    /// </summary>
    public TimeSpan ExecutionDuration { get; set; }

    /// <summary>   
    /// Gets or sets the AnalysisPrompt object used for prompting analysis.
    /// </summary>
    public AnalysisPrompt Prompt { get; set; } = default!;

    /// <summary>
    /// Gets or sets prompt category Id
    /// </summary>
    [NotMapped]
    public Guid PromptCategoryId { get; set; }
}

File: src/Insight.Box.FrontEnd/src/modules/prompts/models/PromptExecutionHistory.ts
/**
 * Represents a data transfer object (DTO) for a prompt execution history record.
 */
export class PromptsExecutionHistory {
    /**
     * Unique identifier of the prompt execution history record.
     */
    public id!: string; // Use 'string' for GUIDs in TypeScript

    /**
     * Unique identifier of the associated prompt.
     */
    public promptId!: string;

    /**
     * The result of the prompt execution (optional).
     */
    public result?: string | null;

    /**
     * Any exception that occurred during the prompt execution (optional).
     */
    public exception?: string | null;

    /**
     * The timestamp of when the prompt execution occurred.
     */
    public executionTime!: Date;

    /**
     * The duration of the prompt execution in seconds.
     */
    public executionDurationInMilliseconds!: number;
}
